File: problems/HackerRank/miniMaxSum.ts
function miniMaxSum(arr: number[]): void {
  // Write your code here
  const sortedArray = arr.sort();
  let minSum = 0;
  let maxSum = 0;

  for (let i = 0; i < arr.length; i++) {
    if (i !== arr.length - 1) {
      minSum += arr[i];
    }
    if (i !== 0) {
      maxSum += arr[i];
    }
  }
  console.log(minSum, maxSum);
}

File: problems/HackerRank/week1/miniMaxSum.py
def miniMaxSum(arr):
    arr.sort()
    
    max = sum(arr[i] for i in range(len(arr)) if i > 0)
    min = sum(arr[i] for i in range(len(arr)) if i < len(arr) - 1)
    
    print(min, max)
    
    
  
    
  # better approach
def miniMaxSum2(arr):
    total = sum(arr)

    min_sum = total - max(arr)    
    max_sum = total - min(arr)
    
    print(min_sum, max_sum)